File: online-judge-solutions/Codeforces/1696A.py
#https://codeforces.com/problemset/problem/1696/A
#Lang		:	Python 3.8
#Time		:	61 ms
#Memory         :	300 KB
for _ in range(int(input())):
    n,z=map(int,input().split())
    s=0
    v=list(map(int,input().split()))[:n]
    for i in v:s=max(i|z,s)
    print(s)

File: online-judge-solutions/Codeforces/1696A.cpp
//https://codeforces.com/problemset/problem/1696/A
//Lang		:	GNU C++14
//Time		:	31 ms
//Memory	:	200 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int t,z,n;
	cin>>t;
	while(t--){
		cin>>n>>z;
		vector<int> v(n);
		int s=0;
		for(int i=0;i<n;i++){
			cin>>v[i];
			s=max(v[i]|z,s);
		}
		cout<<s<<endl;
	}
	return 0;
}
